### PR TITLE
Update `update_extension.json` schema following the latest design changes

### PR DESCRIPTION
**Pull request by karreiro**
### WHY are these changes introduced?

Fixes part of https://github.com/Shopify/shopify/issues/428094

### WHAT is this pull request doing?

This PR just introduces changes generated by the `cd packages/theme && p schema:generate` command. Also, this PR updates unit tests accordingly.

### How to test your changes?

N/A

### Post-release steps

None.

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [x] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Comment by karreiro** [comment]
(this changes are not public-facing, so I'm not adding any `.changeset` entry)

File: packages/theme/src/cli/services/update/check.test.ts [hunk 1]
```diff
@@ -132,6 +132,66 @@ describe('check', () => {
...
+      // When
+      // eslint-disable-next-line no-await-in-loop
+      const errors = await checkErrors(script)
+
+      // Then
+      expect(errors).toEqual([`The "operations/0/id" property is required`])
+    }
+  })
+
+  test(`when operation an operation doesn't have the action property`, async () => {
```
Comment (bahar-p) [inline packages/theme/src/cli/services/update/check.test.ts:167]:
```suggestion
  test(`when an operation doesn't have the action property`, async () => {
```

File: packages/theme/src/cli/services/update/check.test.ts [hunk 2]
```diff
@@ -132,6 +132,66 @@ describe('check', () => {
     expect(errors).toEqual(['The "operations" array must contain at least 1 element(s)'])
   })
 
+  test(`when operation an operation doesn't have the id property`, async () => {
```
[the remark this answers is not in the document]
Comment (karreiro) [reply packages/theme/src/cli/services/update/check.test.ts:135]: Great catch 🚀 Thank you!

File: packages/theme/src/cli/services/update/check.test.ts (hunk 1 above)
Comment (karreiro) [reply packages/theme/src/cli/services/update/check.test.ts:167]: Thank you!

**Pull request merged**